### PR TITLE
Renaming OptimizationTolerance to fix a spelling error.

### LINKED ISSUE #3198: OptmizationTolerance public trainer API options misspelling
Should be "Optim" instead of "Optm". Misspelling occurs on the public API for some trainers, like
https://github.com/dotnet/machinelearning/blob/b6c5b703de4bdc2e0212597075223f4710dc98da/src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs#L54

### PR DESCRIPTION
**Pull request by rogancarr**
In the `LbfgsTrainerBase`, the `OptmizationTolerance` option was misspelled.

Fixes #3198

### CODE REVIEW

**Comment by artidoro** [comment]
Just need to update the entrypoint catalog.

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LbfgsLogisticRegressionWithOptions.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public static void Example()
             var options = new LbfgsLogisticRegressionBinaryTrainer.Options()
             {
                 MaximumNumberOfIterations = 100,
-                OptmizationTolerance = 1e-8f,
+                OptimizationTolerance = 1e-8f,
```
Comment (TomFinley) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LbfgsLogisticRegressionWithOptions.cs]: So, this is desirable, but it is a breaking change, not sure what our procedure and thoughts are about that.

Comment (rogancarr) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LbfgsLogisticRegressionWithOptions.cs]: I was going to see if we could Ask it into 1.0. I will ask.

Comment (rogancarr) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LbfgsLogisticRegressionWithOptions.cs]: Process: Check into master: Cherry-pick into the release branch. PR into release branch will be part of ask-mode.

**Pull request merged**